lib/ansible/modules/cloud/amazon/iot_thing.py

```diff
@@ -0,0 +1,484 @@
+
+- name: add attributes to an IOT thing
+  iot_thing:
+    name: mything
+    attributes:
+      key1: value1
+      key2: value2
+
+- name: remove attributes from an IOT thing
+  iot_thing:
+    name: mything
+    absent_attributes:
+      - key3
+'''
+
+
+RETURN = '''
+default_client_id:
+    description: The ID of the default client for this IOT thing.
+    returned: on success when I(state=present)
+    type: string
+    sample: mythingid
+version:
+    description: The current version of the thing record in the IOT registry.
+    returned: on success when I(state=present)
+    type: int
+    sample: 123
+attribute:
+    description: The attributes currently set on the IOT thing.
+    returned: on success when I(state=present)
+    type: dict
+    sample: {"attribute1": "value1", "attribute2", "value2"}
+principals:
+    description: List of certificate or credential ARNs currently attached to this IOT thing.
+    returned: on success when I(state=present)
+    type: list
+    sample: ["arn:aws:iot:us-east-1:account-id:cert/uuid-1", "arn:aws:iot:us-east-1:account-id:cert/uuid-2"]
+thing_arn:
+    description: The ARN of this IOT thing.
+    returned: on success when the IOT thing is initially created.
+    type: string
+    sample: "arn:aws:iot:us-east-1:account-id:cert/mything"
+thing_name:
+    description: The name of this IOT thing.
+    returned: on success when I(state=present)
+    type: string
+    sample: mything
+thing_type_name:
+    description: The name of this IOT thing.
+    returned: on success when I(state=present) and there is an IOT type associated with this IOT thing.
+    type: string
+    sample: mytype
+'''
+
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import get_aws_connection_info, boto3_conn, camel_dict_to_snake_dict
+
+
+try:
+    from botocore.exceptions import ClientError, ValidationError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
+    argument_spec = dict(
+        name=dict(required=True, type='str'),
+        state=dict(default='present', choices=['present', 'absent']),
+        type=dict(type='str'),
+        remove_type=dict(type='bool'),
+        attached_principals=dict(type='list', default=[]),
+        detached_principals=dict(type='list', default=[]),
+        attributes=dict(type='dict', default=dict()),
+        absent_attributes=dict(type='list', default=[]),
+        expected_version=dict(type='int')
+    )
+
+    mutually_exclusive = [['type', 'remove_type']]
+
+    aws_module = AnsibleAWSModule(argument_spec=argument_spec,
+                                  supports_check_mode=True,
+                                  mutually_exclusive=mutually_exclusive)
+
+    state = aws_module.params.get('state')
+    thing_name = aws_module.params.get('name')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(aws_module, boto3=True)
+    if not region:
```

Comment:
This should no longer be required - boto3_conn will catch a NoRegionError and fail there.